loader/lightning.go

```diff
@@ -0,0 +1,273 @@
+	"github.com/pingcap/dm/dm/pb"
+	"github.com/pingcap/dm/dm/unit"
+	"github.com/pingcap/dm/pkg/binlog"
+	"github.com/pingcap/dm/pkg/conn"
+	tcontext "github.com/pingcap/dm/pkg/context"
+	"github.com/pingcap/dm/pkg/log"
+	"github.com/pingcap/dm/pkg/utils"
+
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tidb/br/pkg/lightning"
+	lcfg "github.com/pingcap/tidb/br/pkg/lightning/config"
+	"go.etcd.io/etcd/clientv3"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+)
+
+const (
+	lightningCheckpointFile  = "lightning.checkpoint.0.sql"
+	lightningCheckpointDB    = "lightning"
+	lightningCheckpointTable = "checkpoint"
+)
+
+// LightningLoader can load your mydumper data into TiDB database.
+type LightningLoader struct {
+	sync.RWMutex
+
+	cfg             *config.SubTaskConfig
+	cli             *clientv3.Client
+	checkPoint      CheckPoint
+	workerName      string
+	logger          log.Logger
+	core            *lightning.Lightning
+	toDB            *conn.BaseDB
+	toDBConns       []*DBConn
+	lightningConfig *lcfg.GlobalConfig
+
+	finish atomic.Bool
+	closed atomic.Bool
+	cancel context.CancelFunc // for per task context, which maybe different from lightning context
+}
+
+// NewLightning creates a new Loader importing data with lightning.
+func NewLightning(cfg *config.SubTaskConfig, cli *clientv3.Client, workerName string) *LightningLoader {
+	lightningCfg := makeGlobalConfig(cfg)
+	core := lightning.New(lightningCfg)
+	loader := &LightningLoader{
+		cfg:             cfg,
+		cli:             cli,
+		core:            core,
+		lightningConfig: lightningCfg,
+		logger:          log.With(zap.String("task", cfg.Name), zap.String("unit", "lightning-load")),
+		workerName:      workerName,
+	}
+	return loader
+}
+
+func makeGlobalConfig(cfg *config.SubTaskConfig) *lcfg.GlobalConfig {
+	lightningCfg := lcfg.NewGlobalConfig()
+	if cfg.To.Security != nil {
+		lightningCfg.Security.CAPath = cfg.To.Security.SSLCA
+		lightningCfg.Security.CertPath = cfg.To.Security.SSLCert
+		lightningCfg.Security.KeyPath = cfg.To.Security.SSLKey
+	}
+	lightningCfg.TiDB.Host = cfg.To.Host
+	lightningCfg.TiDB.Psw = cfg.To.Password
+	lightningCfg.TiDB.User = cfg.To.User
+	lightningCfg.TiDB.Port = cfg.To.Port
+	lightningCfg.TiDB.StatusPort = cfg.TiDB.StatusPort
+	lightningCfg.TiDB.PdAddr = cfg.TiDB.PdAddr
+	lightningCfg.TikvImporter.Backend = cfg.TiDB.Backend
+	lightningCfg.PostRestore.Checksum = lcfg.OpLevelOff
+	if cfg.TiDB.Backend == lcfg.BackendLocal {
+		lightningCfg.TikvImporter.SortedKVDir = cfg.Dir
+	}
+	lightningCfg.Mydumper.SourceDir = cfg.Dir
+	return lightningCfg
+}
+
+// Type implements Unit.Type.
+func (l *LightningLoader) Type() pb.UnitType {
+	return pb.UnitType_Load
+}
+
+// Init initializes loader for a load task, but not start Process.
+// if fail, it should not call l.Close.
+func (l *LightningLoader) Init(ctx context.Context) (err error) {
+	tctx := tcontext.NewContext(ctx, l.logger)
+	checkpoint, err := newRemoteCheckPoint(tctx, l.cfg, l.checkpointID())
```

Comment:
lightning will use file checkpoint by default, and this checkpoint is only a dummy entry? 
How about we use a mysql checkpoint, even if task is failed, user still can query the progress from checkpoint.

we need to remove file checkpoint when enter https://github.com/pingcap/dm/blob/487a3d10f0bb1548ce1c8a0207a1ad4a95fd9f6e/dm/master/server.go#L467